source/MissionAction.cpp

```diff
@@ -385,6 +392,25 @@ bool MissionAction::CanBeDone(const PlayerInfo &player, const shared_ptr<Ship> &
 			return false;
 	}
 	
+	for(const auto &it : requiredAttributes)
+	{
+		double attribute = 0;
+		
+		// Requiring the player to have 0 of this attributes means all ships must be
+		// checked, even if the ship is disabled, parked, or out-of-system.
+		if(!it.second)
+		{
+			for(const auto &ship : player.Ships())
+				if(!ship->IsDestroyed() && ship->Attributes().Get(it.first))
+					return false;
+		}
+		else
+			attribute += flagship ? flagship->Attributes().Get(it.first) : 0;
+		
+		if(attribute < it.second)
```

Comment:
This check won't work correctly for negative `requiredAttributes`

Reply:
How so? Are you expecting a negative required attribute value to return true if the player has an even more negative attribute value?

Reply:
This makes me wonder if instead of this handling like outfits, we instead treat the attributes as we do conditions like reputation, e.g.

```bash
attributes [<location specifier>]
    <attribute name | attribute value expression> <comparison operator> <value | value expression>
    ...
```
These could then be combined logically as is done with conditions.

Reply:
That implementation is on you.

Reply:
This PR does what I had in mind for the issue, so I think this is fine to merge as is, and the mechanic can be expanded upon later. If you think it should be the case though, I can change it so that negative attribute will compare for greater to return false than while positive values still compare for less than.